### DISCUSSION TITLE
Plot support on low timeframes

### DISCUSSION THREAD

**Discussion by Kayne88** [kernc/backtesting.py#499] [General]
Hello,

I was wondering, if there is a way for the plot method to support low timeframes (like 1m, 2m, 3m candlestick charts). When I was trying it, the html ran into performance problems. Also, I dont want to resample the data, since I will loose information on my trades.

cheers

**Comment by kernc** [comment]
The performance problems are due to Bokeh, JavaScript, and your browser environment. You can play around with [`Backtest.plot(..., resample=)`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.plot) parameter, which defaults to outputting 10k candles. Other than that, you can get a better computer, or plot/inspect visualizations piece-wise. Maybe also try [accelerating Bokeh with WebGL](https://docs.bokeh.org/en/latest/docs/user_guide/webgl.html).

**Reply by CadeHalcyon** [reply]
Hi, regarding the issue outlined above, I have 1 mins candle data over 6 months... setting resample=False just about works, but laggy performance issues navigating the chart output.  Question:  when it resamples from say 1 mins to display 30 mins candles (as in my case with resample set to True), are the trades still based off of the 1 min candle source data?  i.e. the trades can enter and exit within the 30 mins candles if trade entry/exit criteria are met on the 1 min timeframe source dataframe?

**Reply by kernc** [reply]
The laggy performance comes from bokeh/JavaScript/browser, so you can set on your own higher `resample='4H'`, `resample='1D'` or similar. The backtesting/trading is always done on the data timeframe (i.e. 1 minute). `Backtest.plot(resample=...)` is about plotting only.

**Reply by CadeHalcyon** [reply]
Perfect, thanks!

**Reply by CadeHalcyon** [reply]
Hi, further to the above, I am experiencing issues with trade entries, and it seems it is to do with the issues outlined above.

In my dataframe, I manually add 3 columns after the Volume column.  Each column has a $value fixed for 24hrs or 1,440 mins rows and there is a new value at 1am each day (when UK was UTC+1, and now 12am now we are UTC).

The trades I enter are based on these indicator values, but only one trade per day, at daily open.

Issues I am encountering:

1) Using 1 mins candle data for my dataframe over 6 months, the chart with resampling set to True shows 30mins candles.  I plot my indicators as lines on the chart, but for some reason they change at 1:30am each day, rather than aligning to the dataframe which updates the indicator values at 1am each day.  Despite this, my trading strategy seems to recognise the values in the dataframe rather than the chart.  I have no issue with this as the trades are based off the dataframe values.

HOWEVER...

2). I tried the same over 12 months with 1mins chart data, and the chart resamples to 1hr candles.  Now each indicator is not reflected in the lines on the chart until 2am each day, instead of 1am in the dataframe.  Now the trades are also wrong - same trade strategy, but now when the trade enters at 1am, it is using the previous day's indicator values, i.e. seemingly ignoring the dataframe and using the chart's values instead.

Hope this makes sense, do let me know if I should support this with screenshots / code etc.  but started without as thought this might be a known issue with a resolution.  Thanks!
